src/main/java/com/minecolonies/coremod/network/messages/HutRenameMessage.java

```diff
@@ -0,0 +1,110 @@
+import io.netty.buffer.ByteBuf;
+import net.minecraft.entity.player.EntityPlayerMP;
+import net.minecraft.util.math.BlockPos;
+import net.minecraftforge.fml.common.network.ByteBufUtils;
+import net.minecraftforge.fml.common.network.simpleimpl.IMessage;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Message to execute the renaiming of the townHall.
+ */
+public class HutRenameMessage extends AbstractMessage<HutRenameMessage, IMessage>
+{
+    /**
+     * The max length of the name.
+     */
+    private static final int MAX_NAME_LENGTH = 25;
+
+    /**
+     * Max substring length.
+     */
+    private static final int SUBSTRING_LENGTH = MAX_NAME_LENGTH - 1;
+
+    /**
+     * The colony id.
+     */
+    private int colonyId;
+
+    /**
+     * The custom name to set.
+     */
+    private String name;
+
+    /**
+     * The building id.
+     */
+    private BlockPos buildingId;
+
+    /**
+     * The dimension of the message.
+     */
+    private int dimension;
+
+    /**
+     * Empty public constructor.
+     */
+    public HutRenameMessage()
+    {
+        super();
+    }
+
+    /**
+     * Object creation for the town hall rename message.
+     *
+     * @param colony Colony the rename is going to occur in.
+     * @param name   New name of the town hall.
+     */
+    public HutRenameMessage(@NotNull final IColony colony, final String name, final AbstractBuildingView b)
+    {
+        super();
+        this.colonyId = colony.getID();
+        this.name = (name.length() <= MAX_NAME_LENGTH) ? name : name.substring(0, SUBSTRING_LENGTH);
+        this.dimension = colony.getDimension();
+        this.buildingId = b.getID();
+    }
+
+    @Override
+    public void fromBytes(@NotNull final ByteBuf buf)
+    {
+        colonyId = buf.readInt();
+        name = ByteBufUtils.readUTF8String(buf);
+        dimension = buf.readInt();
+        buildingId = BlockPosUtil.readFromByteBuf(buf);
+    }
+
+    @Override
+    public void toBytes(@NotNull final ByteBuf buf)
+    {
+        buf.writeInt(colonyId);
+        ByteBufUtils.writeUTF8String(buf, name);
+        buf.writeInt(dimension);
+        BlockPosUtil.writeToByteBuf(buf, buildingId);
+    }
+
+    @Override
+    public void messageOnServerThread(final HutRenameMessage message, final EntityPlayerMP player)
+    {
+        final Colony colony = ColonyManager.getColonyByDimension(message.colonyId, message.dimension);
+        if (colony != null && colony.getPermissions().hasPermission(player, Action.MANAGE_HUTS))
+        {
+            final AbstractBuilding b = colony.getBuildingManager().getBuildings().get(message.buildingId);
+
+            if (b != null)
+            {
+                message.name = (message.name.length() <= MAX_NAME_LENGTH) ? message.name : message.name.substring(0, SUBSTRING_LENGTH);
+                b.setCustomBuildingName(message.name);
+                MineColonies.getNetwork().sendToAll(message);
```

Comment:
this message does not seem to have a client acceptor method to be handled by messageOnClientThread

Reply:
ugh, copy and paste error

Reply:
Uhm, it should only be handled on server side and not on client side

Reply:
This one